r-package/grf/tests/testthat/test_boosted_regression_forest.R

```diff
@@ -21,22 +21,46 @@ test_that("Boosted regression forest improves performance vs regular forest", {
 
 
 test_that("Boosted forest takes user specified number of steps",  {
-  n = 100; p = 6
-  X = matrix(runif(n * p), n, p)
-  mu = 2 * X[,1] * X[,2] + 3 * X[,3] + 4 * X[,4]
-  Y = mu + rnorm(n)
+  n <- 100; p <- 6
+  X <- matrix(runif(n * p), n, p)
+  mu <- 2 * X[,1] * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y <- mu + rnorm(n)
   forest.boost <- boosted_regression_forest(X,Y,boost.steps=2)
   expect_equal(2,length(forest.boost$forests))
 })
 
- test_that("Under cross-validation, errors decrease each step", {
-   n=200; p=6
-   X = matrix(runif(n * p), n, p)
-   mu = 2 * X[,1]^2 * X[,2] + 3 * X[,3] + 4 * X[,4]
-   Y = mu + rnorm(n)
-   forest.boost <- boosted_regression_forest(X,Y)
-   errors <- unlist(lapply(forest.boost$debiased.errors,mean))
-   improves <- errors[2:length(errors)] - errors[1:(length(errors)-1)]
-   expect_true(all(improves>0))
-
- })
+test_that("Under cross-validation, errors decrease each step", {
+  n<-200; p<-6
+  X <- matrix(runif(n * p), n, p)
+  mu <- 2 * X[,1]^2 * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y <- mu + rnorm(n)
+  forest.boost <- boosted_regression_forest(X,Y)
+  errors <- unlist(forest.boost$error)
+  if(length(errors)>1) {
+    improves <- errors[2:length(errors)] - errors[1:(length(errors)-1)]
+    expect_true(all(improves<0))
+  }
+})
+
+test_that("boost.error.reduction validation works", {
+  n<-200; p<-6
+  X <- matrix(runif(n * p), n, p)
+  mu <- 2 * X[,1]^2 * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y <- mu + rnorm(n)
+  expect_error(forest.boost <- boosted_regression_forest(X,Y,boost.error.reduction=1.5))
+})
+
+test_that("OOB prediction is close to actual out of sample error", {
+  n<-3000; p<-6
+  X <- matrix(runif(n * p), n, p)
+  mu <- 2 * X[,1]^2 * X[,2] + 3 * X[,3] + 4 * X[,4]
+  Y <- mu + rnorm(n)
+  test <- 2000:3000
+  train <- 1:2000
+  forest.boost <- boosted_regression_forest(X[train,],Y[train],boost.steps=2)
+  OOB_error <- unlist(forest.boost$error)[length(forest.boost$error)]
```

Comment:
Could we use the actual OOB predictions here (what's returned from `predict(forest.boost)`), as that is what users will primarily be working with?

Also small comment: variable names in R should use dots instead of underscores.

Reply:
This is fixed now.